### PR TITLE
[SPE-6] Boundary vs. other software

### PR DESCRIPTION
**Pull request by Dan-Heath**
Per [SPE-6](https://hashicorp.atlassian.net/browse/SPE-6), the Boundary vs other software page should philosphically compare Boundary to other solutions.

~~[View an updated preview here.](https://boundary-g0efqywmx-hashicorp.vercel.app/boundary/docs/overview/other-software)~~

NOTE: ~~I am having some issues with the .json for the navigation. we wanted to nest **Use Cases** and **Boundary vs. Other Software** under **What is Boundary?**, but I can't move that topic b/c it has dependencies in other repos. I am still working out how to handle this.~~

Navigation is fixed now. I created a new bucket - **What is Boundary?** and added the topics **What is Boundary?**, **Use Cases**, and **Boundary vs. Other Software** below it. I renamed the **What is Boundary?** topic **Overview** in the navigation pane:

<image>

EDIT: I split each of the sections from this topic out into a standalone topic to make this easier to read and improve SEO. [New preview is here.](https://boundary-roev5ywsl-hashicorp.vercel.app/boundary/docs/overview/vs/other-software)

### CODE REVIEW

File: website/content/docs/overview/other-software.mdx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+Boundary is a software solution that provides secure remote access to infrastructure services.
+It can be difficult to understand how to compare the different remote access security solutions that are available on the market.
+Is one a replacement for another?
+Are they complementary?
+In this topic, we compare Boundary to other available security solutions.
+
+## Introduction
+
+The following comparisons are philosophical in nature, and are not intended to be opinionated.
+There are many possible remote access use cases, and the following solutions could have some part in your overall security strategy.
```
Comment (Dan-Heath) [inline website/content/docs/overview/other-software.mdx:18]: I added this in response to Jairo's comment from the doc about removing the word "negative." But I'm not sure if "opinionated" is the best word. I am open to any suggestions. Thanks!

Comment (AdamBouhmad) [reply website/content/docs/overview/other-software.mdx:18]: Perhaps just `The following comparisons are philosophical in nature, and are not intended to be oppositional.`

File: website/content/docs/overview/other-software.mdx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+You can use Boundary and other HashiCorp tools such as Vault and Consul to create a zero trust access model for infrastructure endpoints in which all transactions are authenticated, authorized, and audited.
+Boundary focuses on giving developers a consistent workflow for zero trust access to their infrastructure, wherever it resides.
+
+## Boundary vs. Bastion Hosts
+
+If you want to set up your cloud environment securely, you may choose to run all of your important workloads behind a NAT Gateway, and provision a DMZ with a set of hardened bastion servers.
+
+Bastion host security groups are often not locked down at the network layer.
+Additionally, users who log into a bastion host using SSH are typically dropped into a privileged account.
+Maintaining the security groups, network ACLs, and IAM control on the bastion host at a per-user level is nearly impossible, unless you create and maintain multiple bastion hosts per user or group.
```
Comment (AdamBouhmad) [inline website/content/docs/overview/other-software.mdx:43]: ``
```suggestion
Maintaining security groups, network ACLs, and IAM controls on a bastion host at a per-user level is nearly impossible, unless you create and maintain multiple bastion hosts per user or group.
```

File: website/content/docs/overview/other-software.mdx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+IT departments now have to manage updates for another server, and the sprawl of infrastructure continues, increasing your attack surface, and requiring your IT department to be perfect.
```
Comment (AdamBouhmad) [inline website/content/docs/overview/other-software.mdx:44]:
```suggestion
IT departments now have to manage updates for another server and the sprawl of infrastructure continues, increasing your attack surface, and requiring your IT department to be perfect.
```

File: website/content/docs/overview/other-software.mdx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+A traditional VPN provides users with **network access**.
+Boundary provides users with **delegated access to services within the network**.
+
+_Can Boundary replace a VPN?_
+
+For many organizations, Boundary's granular network access will be a security improvement from traditional data center VPN solutions that lack granular controls.
+
+_Can Boundary work with a VPN?_
+
+Boundary can work with an existing corporate VPN to provide heightened security when accessing privileged networks such as data centers and cloud VPCs.
```
Comment (AdamBouhmad) [inline website/content/docs/overview/other-software.mdx:85]:
```suggestion
Boundary can work with existing corporate VPNs to provide heightened security when accessing privileged networks such as data centers and cloud VPCs.
```

File: website/content/docs/overview/other-software.mdx (hunk 1 above)
Comment (Dan-Heath) [reply website/content/docs/overview/other-software.mdx:18]: > Perhaps just `The following comparisons are philosophical in nature, and are not intended to be oppositional.`

Nice, I like that.

**Comment by Dan-Heath** [comment]
> Few updates, otherwise LGTM! Awesome job @Dan-Heath

Thanks @AdamBouhmad ! I appreciate the feedback. I made those updates and the new preview is [at this link](https://boundary-g0efqywmx-hashicorp.vercel.app/boundary/docs/overview/other-software).

**Review by AdamBouhmad — APPROVED** [review]
Looks great!

**Comment by stellarsquall** [comment]
I overall really like the tone of this document, and appreciate that it attempts to clarify where Boundary fits in comparison to other access model solutions. I fully approve of this as a first draft 👍 

In general, I feel like I'm confronted with a large wall of text when I visit this page. Is there a way we can consider adding a comparison table to show what Boundary is, and is not? Perhaps by doing a feature comparison to the other solutions listed on this page (bastion hosts, VPNs, PAM, software defined perimeter, secrets management tools)? I'm thinking about something visually resembling the [Vault interoperability matrix](https://developer.hashicorp.com/vault/docs/interoperability-matrix), to visually show what boundary is and is not

Another option would be to just break up the text with diagrams or other content that is visually appealing. But I'm left feeling like it takes effort to read to the bottom of the page to get the clarity I'm looking for

**Comment by Dan-Heath** [comment]
> In general, I feel like I'm confronted with a large wall of text when I visit this page. 

Thank you for the review Robin! Yep, I can see this; it's a lot of content. Adam pointed out that in the [Waypoint version](https://developer.hashicorp.com/waypoint/docs/intro/vs), each of the individual technologies is a standalone topic and they're all nested together. So I just updated this one with the same format. I think this should make it easier to read (and better SEO too):

[New preview deployment
](https://boundary-roev5ywsl-hashicorp.vercel.app/boundary/docs/overview/vs/other-software)

> Is there a way we can consider adding a comparison table to show what Boundary is, and is not? Perhaps by doing a feature comparison to the other solutions listed on this page (bastion hosts, VPNs, PAM, software defined perimeter, secrets management tools)? I'm thinking about something visually resembling the [Vault interoperability matrix](https://developer.hashicorp.com/vault/docs/interoperability-matrix), to visually show what boundary is and is not

I love this idea as a next step for these topics. I think that table would work really well as a high level comparison on the initial Overview page, and then users could drill down to the individual topics for more info.

**Comment by stellarsquall** [comment]
I like this format more, and I think it gives the individual comparison pages room to grow 👍  The next step for this comparison section could be tables like I mentioned above, or diagrams that make the conceptual content for each page clearer.

Thanks for taking this approach! @AdamBouhmad should probably sign off on the new format

**Comment by Dan-Heath** [comment]
NOTE: Since this PR nests the **What is Boundary** topic under the _Introduction_ heading, I had to create a PR to update incoming links from the `dev-portal` repo. The following PR should be merged with this one to ensure we don't break any links:

https://github.com/hashicorp/dev-portal/pull/1383

**Review by stellarsquall — APPROVED** [review]

**Pull request merged**